membership-attribute-service/app/controllers/PaymentDetailMapper.scala

```diff
@@ -33,11 +33,11 @@ object PaymentDetailMapper {
   )
 
   def paymentDetailsForSub(
-    maybeUserId: Option[String],
+    isGiftRedemption: Boolean,
     freeOrPaidSub: Subscription[SubscriptionPlan.Free] \/ Subscription[SubscriptionPlan.Paid],
     paymentService: PaymentService
   ): Future[PaymentDetails] = freeOrPaidSub match {
-    case \/-(giftSub) if giftSub.gifteeIdentityId == maybeUserId && giftSub.readerType == Gift =>
+    case \/-(giftSub) if giftSub.readerType == Gift && isGiftRedemption =>
```

Comment:
```suggestion
    case \/-(giftSub) if isGiftRedemption =>
```

Reply:
wouldn't that be enough - we already know it's a gift as it has the giftee identity id filled in? https://github.com/guardian/membership-common/blob/74ec1966b3393d3eb2b86b8c247ac875ffc4b254/src/main/scala/com/gu/zuora/rest/ZuoraRestService.scala#L405
`select name, id, termEndDate from subscription where GifteeIdentityId__c = '${identityId}' and status = 'Active' and termEndDate >= '$today'`